GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -320,6 +327,98 @@ public void run() {
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render +y face
+        centerCameraTransform.rotateByAxis(-90, 0, 1, 0);
+        centerCameraTransform.rotateByAxis(90, 1, 0, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render -y face
+        centerCameraTransform.rotateByAxis(180, 1, 0, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render +z face
+        centerCameraTransform.rotateByAxis(90, 1, 0, 0);
+        centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        // render -z face
+        centerCameraTransform.rotateByAxis(180, 0, 1, 0);
+        renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                mRenderBundle.getRightRenderTexture(), mRenderBundle);
+        readRenderResult(mRenderBundle.getPostEffectRenderTextureA().getPtr(),
+                readbackBuffer);
+        byteArrayList.add(Arrays.copyOf(readbackBuffer.array(),
+                readbackBuffer.array().length));
+
+        centerCameraObject.detachCamera();
+        mainCameraRig.getOwnerObject().removeChildObject(centerCameraObject);
+    }
+
+    void returnScreenshot3DToCaller(final GVRScreenshot3DCallback callback,
+            final List<byte[]> byteArrayList, final int width, final int height) {
+        if (byteArrayList.size() != 6) {
+            throw new IllegalArgumentException("byteArrayList length is not 6.");
+        } else {
+            // run the callback function in a background thread
+            Threads.spawn(new Runnable() {
+                public void run() {
+                    List<Bitmap> bitmapList = new ArrayList<Bitmap>();
+                    for (byte[] bytearray : byteArrayList) {
```

Comment:
You know, some of our devices do have 8 cores, these days. Doing a `Threads.spawnLow()` for each face and waiting for all 6 to finish is going to be up to 6x faster than looping. I don't think call-to-callback time is incredibly important to the app - but there _is_ benefit to releasing these large byte arrays quickly. Minimizes heap fragmentation and all that.